src/hotspot/share/logging/logFileStreamOutput.cpp

```diff
@@ -146,6 +164,28 @@ int LogFileStreamOutput::write_internal(const LogDecorations& decorations, const
   return written;
 }
 
+int LogFileStreamOutput::write_internal(const LogDecorations& decorations, const char* msg) {
+  int msg_len = (int)strlen(msg);
+
+  // Do not do anything if foldmultilines has been specified
```

Comment:
But we do something, right? Maybe "don't alter the string if _fold_multilines has been specified". Please add a corresponding explanation to the other case, probably something mentioning "canonical representation" or something of the sort.